Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift

```diff
@@ -123,3 +123,16 @@ public extension SwitchStmtSyntax {
     self.cases = casesBuilder()
   }
 }
+
+// MARK: - VariableDeclSyntax
+// VariableDeclSyntax is a special scenario as it don't have body or members
+// So we cannot conform to `HasTrailingCodeBlock` or `HasTrailingMemberDeclBlock`
+
+public extension VariableDeclSyntax {
+  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder accessor: () -> CodeBlockItemListSyntax) {
+    self = "\(signature) {}"
+    var binding: PatternBindingSyntax? = self.bindings.last
+    binding?.accessor = .getter(CodeBlockSyntax(statements: accessor()))
+    bindings = PatternBindingListSyntax([binding].compactMap { $0 })
```

Comment:
I would add an `assert(self.bindings.count == 1)` after `self = "\(signature) {}" ` because we would be dropping all but the last binding here.